packages/core/src/components/fields/NumberField.tsx

```diff
@@ -34,30 +30,20 @@ type NumberFieldState = {
  *    value cached in the state. If it matches the cached value, the cached
  *    value is passed to the input instead of the formData value
  */
-class NumberField<T = any, F = any> extends Component<
-  FieldProps<T, F>,
-  NumberFieldState
-> {
-  /** Constructs the `NumberField` from the `props`
-   *
-   * @param props - The `FieldProps` for this component
-   */
-  constructor(props: FieldProps<T, F>) {
-    super(props);
+function NumberField<T = any, F = any>(props: FieldProps<T, F>) {
+  const { registry, onChange, formData, value: initialValue } = props;
+  const [lastValue, setLastValue] = useState(initialValue);
+  const { StringField } = registry.fields;
 
-    this.state = {
-      lastValue: props.value,
-    };
-  }
+  let value = formData;
 
   /** Handle the change from the `StringField` to properly convert to a number
    *
    * @param value - The current value for the change occurring
    */
-  handleChange = (value: any) => {
-    const { onChange } = this.props;
+  const handleChange = (value: FieldProps<T, F>["value"]) => {
     // Cache the original value in component state
-    this.setState({ lastValue: value });
+    setLastValue(value);
```

Comment:
consider using `useRef` instead to avoid an additional rerender

Reply:
Hi Jimmy, I attempted your suggestion, but the tests failed so I rolled back to using state.